src/core_plugins/kibana/public/context/api/context.js

```diff
@@ -0,0 +1,52 @@
+import _ from 'lodash';
+
+import { addComputedFields } from './utils/fields';
+import { getDocumentUid } from './utils/ids';
+import { createSuccessorsQuery } from './utils/queries.js';
+import { reverseQuerySort } from './utils/sorting';
+
+
+async function fetchSuccessors(es, indexPattern, anchorDocument, sort, size) {
+  const successorsQuery = prepareQuery(indexPattern, anchorDocument, sort, size);
+  const results = await performQuery(es, indexPattern, successorsQuery);
+  return results;
+}
+
+async function fetchPredecessors(es, indexPattern, anchorDocument, sort, size) {
+  const successorsQuery = prepareQuery(indexPattern, anchorDocument, sort, size);
+  const predecessorsQuery = reverseQuerySort(successorsQuery);
+  const reversedResults = await performQuery(es, indexPattern, predecessorsQuery);
+  const results = reverseResults(reversedResults);
+  return results;
+}
+
+
+function prepareQuery(indexPattern, anchorDocument, sort, size) {
+  const successorsQuery = addComputedFields(
+    indexPattern,
+    createSuccessorsQuery(anchorDocument.sort, sort, size)
+  );
+  return successorsQuery;
```

Comment:
After some more thought, I think that most of my comments about renaming and organization don't need to block this PR and can be addressed in a later refactoring PR, after I've had some time to put together a more comprehensive proposal.